### PR TITLE
[mtoh] Add playblasting support.

### PR DESCRIPTION
**Pull request by marsupial**

### CODE REVIEW

File: lib/mayaUsd/render/mayaToHydra/renderOverride.cpp [hunk 1]
```diff
@@ -354,7 +363,7 @@ MStatus MtohRenderOverride::Render(const MHWRender::MDrawContext& drawContext) {
     // }
     TF_DEBUG(HDMAYA_RENDEROVERRIDE_RENDER)
         .Msg("MtohRenderOverride::Render()\n");
-    auto renderFrame = [&]() {
+    auto renderFrame = [&](bool markTime = false) {
         const auto originX = 0;
```
Comment (pmolodo) [inline lib/mayaUsd/render/mayaToHydra/renderOverride.cpp:367]: In what situations is renderFrame(false) called?

File: lib/mayaUsd/render/mayaToHydra/renderOverride.cpp [hunk 2]
```diff
@@ -377,8 +386,44 @@ MStatus MtohRenderOverride::Render(const MHWRender::MDrawContext& drawContext) {
...
+            // XXX: Is this better as user-configurable ?
+            constexpr auto msWait = std::chrono::duration<float, std::milli>(100);
+
+            std::shared_ptr<HdxProgressiveTask> progressiveTask =
+                std::dynamic_pointer_cast<HdxProgressiveTask>(tasks.front());
+            if (progressiveTask) {
+                HdTaskSharedPtrVector renderOnly = { progressiveTask };
+                _engine.Execute(_renderIndex, &renderOnly);
+
+                while (!progressiveTask->IsConverged()) {
```
Comment (pmolodo) [inline lib/mayaUsd/render/mayaToHydra/renderOverride.cpp:410]: Is there an easy way that we can add a check to see if the user has requested a cancel to this loop?  Even in normal situations, convergence could take a while, and in worst case, the underlying renderer has some sort of a bug which makes it never properly converge.

Comment (marsupial) [reply lib/mayaUsd/render/mayaToHydra/renderOverride.cpp:410]: Uhhh yeah, I seem to have dropped the **_playBlasting** check in a refactor..

File: lib/mayaUsd/render/mayaToHydra/renderOverride.cpp (hunk 1 above)
Comment (marsupial) [reply lib/mayaUsd/render/mayaToHydra/renderOverride.cpp:367]: I have trouble following the logic, but there's a case below the first renderFrame where Storm is called again (looks like to draw wireframe).

**Comment by marsupial** [comment]
Updated with cancelaltion.

File: lib/mayaUsd/render/mayaToHydra/renderOverride.cpp (hunk 1 above)
Comment (pmolodo) [reply lib/mayaUsd/render/mayaToHydra/renderOverride.cpp:367]: Ah, missed that... makes sense then!

File: lib/mayaUsd/render/mayaToHydra/renderOverride.cpp (hunk 2 above)
Comment (pmolodo) [reply lib/mayaUsd/render/mayaToHydra/renderOverride.cpp]: Awesome, thanks for the fix!

**Comment by kxl-adsk** [comment]
@marsupial Do you have any update on this PR? Would be great to fix compilation error with older USD versions and get it in.

**Comment by marsupial** [comment]
Moving to 20.08, so I am updating it, but not sure how much it's in the cards for support < 20.05 here though...you might be on your own with that.

**Comment by kxl-adsk** [comment]
We won't be able to accept the PR if it doesn't build with supported USD versions.

**Comment by marsupial** [comment]
Understand, but there isn't much need (or resources) to support these old versions on our side.
Hopefully the latest alleviates the issues going back to 20.02

**Pull request merged**